pkg/controller/k8s/endpoints_controller.go

```diff
@@ -168,10 +170,16 @@ func compareServicePorts(new, old map[string]*svc.ServicePort) (add, update, del
 	return
 }
 
+func genSvcPortName(svcName, portName string) string {
+	// #nosec
+	sum := md5.Sum([]byte(svcName + "/" + portName))
+	return hex.EncodeToString(sum[:])
+}
+
 func newSvcPort(epNamespacedName types.NamespacedName, portName string) *svc.ServicePort {
 	return &svc.ServicePort{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      string(uuid.NewUUID()),
+			Name:      genSvcPortName(epNamespacedName.Name, portName),
```

Comment:
Does the name have length limit?

Reply:
md5 to string, length is 32
the length of k8s resource name <= 253
32 is supported